### PR TITLE
Install from lockfile by category

### LINKED ISSUE #1883: Micromamba: allow selecting category for new lockfile format
Based on the lack of command-line options and skimming the source code, it appears that micromamba is hardcoded to install only packages where `category=main`. It would be convenient to, for example, specify both `main` and `dev`.

https://github.com/mamba-org/mamba/blob/e0527eb061b0ce724dcb3c2f34db25c907ee7113/libmamba/src/core/transaction.cpp#L1572

### PR DESCRIPTION
**Pull request by jvansanten**
Add a `--category` flag to restrict which category of package should be installed from an environment lockfile. Defaults to "main".

Closes #1883.

### CODE REVIEW

File: libmamba/src/api/configuration.cpp [hunk 1]
```diff
@@ -1264,6 +1264,10 @@ namespace mamba
                    .group("Solver")
                    .description("Only install dependencies"));
 
+        insert(Configurable("categories", std::vector<std::string>({ "main" }))
+                   .group("Solver")
+                   .description("Package categories to consider"));
```
Comment (jonashaag) [inline libmamba/src/api/configuration.cpp:1269]:
```suggestion
                   .description("Package categories to consider when installing from a lock file"));
```

File: libmamba/src/core/transaction.cpp [hunk 2]
```diff
@@ -1573,31 +1574,45 @@ namespace mamba
...
+        {
+            std::vector<PackageInfo> conda, pip;
+        } packages;
+
+        for (const auto& category : categories)
+        {
+            std::vector<PackageInfo> selected_packages;
+
+            selected_packages = lockfile_data.get_packages_for(
+                category, Context::instance().platform, default_manager);
```
Comment (jonashaag) [inline libmamba/src/core/transaction.cpp:1589]: Let's hardcode "conda" here as well? We're already hardcoding it in the `back_inserter` call.

**Review by jonashaag — COMMENTED** [review]
Thanks, LGTM, although I'm not familiar with this area of code. Can you add a small test?

**Review by jonashaag — APPROVED** [review]
Awesome!

**Comment by wolfv** [comment]
Thanks!

**Pull request merged**